pkg/flowser/app.go

```diff
@@ -0,0 +1,207 @@
+package flowser
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/google/go-github/github"
+)
+
+const (
+	darwin  = "darwin"
+	windows = "windows"
+)
+
+// New create new Flowser application.
+func New() *App {
+	return &App{}
+}
+
+type App struct {
+	installDir string
+}
+
+var errorPlatformNotSupported = errors.New("OS not supported, only supporting Windows and Mac OS")
+
+// Run starts the Flowser application with provided path to the flow project.
+//
+// Project path if exists should be set to the folder containing flow.json, if no such path exists pass empty value.
+func (a *App) Run(installDir string, projectPath string) error {
+	exe, err := a.executable(installDir)
+	if err != nil {
+		return err
+	}
+
+	if projectPath != "" {
+		projectPath = fmt.Sprintf("--project-path=%s", projectPath)
+	}
+
+	return exec.
+		Command(exe, projectPath).
+		Run()
+}
+
+// Install Flowser application in the provided install directory.
+//
+// Install directory is optional, if you want to default to your system location you can provide empty value.
+func (a *App) Install(installDir string) error {
+	downloadDir, err := downloadLatestReleaseAsset()
+	if err != nil {
+		return err
+	}
+
+	defer os.Remove(downloadDir)
+	return a.unzip(downloadDir, installDir)
+}
+
+// Installed checks whether the flowser application is already installed on the system.
+func (a *App) Installed(installDir string) bool {
+	executable, err := a.executable(installDir)
+	if err != nil {
+		return false
+	}
+
+	_, err = os.Stat(executable)
+	return err == nil
+}
+
+// Remove Flowser application from provide directory.
+//
+// Install directory is optional, if you don't provide a value default will be used.
```

Comment:
Seems like "Install directory is optional..." comments are outdated as the default value is set in Flow CLI afaik.